flink-runtime/src/test/java/org/apache/flink/runtime/resourcemanager/slotmanager/DefaultResourceAllocationStrategyTest.java

```diff
@@ -138,6 +137,51 @@ void testFulfillRequirementWithRegisteredResourcesEvenly() {
                                         .isTrue());
     }
 
+    @Test
+    void testFullFillRequirementWithMinimumPossibleRegisteredResources() {
```

Comment:
Should we also test a call, where a TM is already more utilized than a 2nd TM and we verify that the allocation attempt uses the more-utilized TM (essentially, try to come up with a test case where the any strategy fails but the new strategy succeeds).